plutus-contract/src/Plutus/Contract/Test/ContractModel/DoubleSatisfaction.hs

```diff
@@ -30,66 +30,25 @@ module Plutus.Contract.Test.ContractModel.DoubleSatisfaction
     , checkDoubleSatisfactionWithOptions
     ) where
 
-import PlutusTx qualified
-import PlutusTx.Builtins hiding (error)
-
-import Cardano.Api qualified as C
-import Cardano.Api.Shelley qualified as C
-import Control.Lens
-import Control.Monad.Cont
-import Control.Monad.Freer (Eff, run)
-import Control.Monad.Freer.Extras.Log (LogMessage, logMessageContent)
-import Control.Monad.State qualified as State
-import Data.Map qualified as Map
-import Data.Maybe
-
-import Data.Either.Combinators (leftToMaybe)
-
-import Ledger qualified as P
-import Ledger.Ada qualified as Ada
-import Ledger.CardanoWallet qualified as CW
-import Ledger.Crypto
-import Ledger.Generators
-import Ledger.Index as Index
-import Ledger.Scripts
-import Ledger.Slot
-import Ledger.Tx hiding (mint)
-import Ledger.Tx.CardanoAPI (adaToCardanoValue, fromCardanoTxOutToPV1TxInfoTxOut, toCardanoAddressInEra,
-                             toCardanoTxOutDatumInTx, toCardanoTxOutDatumInline)
-import Ledger.Validation qualified as Validation
 import Plutus.Contract.Test hiding (not)
 import Plutus.Contract.Test.ContractModel.Internal
-import Plutus.Trace.Emulator as Trace (EmulatorTrace, activateContract, callEndpoint, runEmulatorStream)
-import Plutus.V1.Ledger.Address
-import Streaming qualified as S
 
 import Test.QuickCheck hiding (ShrinkState, checkCoverage, getSize, (.&&.), (.||.))
-import Test.QuickCheck.Monadic (monadic)
-import Test.QuickCheck.Monadic qualified as QC
-import Test.QuickCheck.StateModel hiding (Action, Actions (..), actionName, arbitraryAction, initialState, monitoring,
-                                   nextState, pattern Actions, perform, precondition, shrinkAction, stateAfter)
-import Test.QuickCheck.StateModel qualified as StateModel
-
-import Wallet.Emulator.Chain hiding (_currentSlot, currentSlot)
-import Wallet.Emulator.MultiAgent (EmulatorEvent, EmulatorEvent' (ChainEvent), eteEvent)
-import Wallet.Emulator.Stream (EmulatorConfig (_params), EmulatorErr)
-
-import Data.Either (fromRight)
-import Prettyprinter
+import Test.QuickCheck.ContractModel
 
 -- Double satisfaction magic
```

Comment:
We are working on a new framework that generalizes `DoubleSatisfaction` to deal with any such properties. We could just drop `DoubleSatisfaction` here and put that stuff in a separate PR - it won't be done for another few weeks.

Reply:
It sounds ok to me. Note that I updated recently (after this PR) `DoubleSatisfaction` so that it uses the `Tx` from cardano-api instead of the ones of `plutus-apps.Ledger` it would be great if the new framework could migrate to them too.
I suggest the following: remove `DoubleSatisfaction` and comment the corresponding tests in `plutus-use-cases` with a note indicating that they should be uncommented once your new framework is available.

Reply:
Yes! The new framework will work with cardano-api and will implement a general tx-level mutation-testing framework that subsumes double satisfaction. The core implementation will live in quickcheck-contractmodel and the layer in plutus-apps will be "thin".